### PR TITLE
Feature/admin user management

### PR DESCRIPTION
**Pull request by AlessioVerardo**
Hello, 
This PR add a way to display the list of all users registered in the database. Furthermore, the admin can now edit the profiles of other users and also promote / demote them. 

One test has also been fixed. 

Cheers

### CODE REVIEW

**Review by Cell00phane — COMMENTED** [review]
Good refactoring of tests organizing them into packages, and of the string values!
Just need to pass the tests.

**Review by Cell00phane — APPROVED** [review]

File: app/src/test/java/com/github/sdpteam15/polyevents/database/objects/UserDatabaseTest.kt [hunk 1]
```diff
@@ -303,4 +312,44 @@ class UserDatabaseTest {
...
+        val testUser = ObservableList<UserEntity>()
+        userToBeAdded.add(UserEntity("uid1","username1","name1"))
+        userToBeAdded.add(UserEntity("uid2","username2","name2"))
+        userToBeAdded.add(UserEntity("uid3","username3","name3"))
+        userToBeAdded.add(UserEntity("uid4","username4","name4"))
+        userToBeAdded.add(UserEntity("uid5","username5","name5"))
+        userToBeAdded.add(UserEntity("uid6","username6","name6"))
+        userToBeAdded.add(UserEntity("uid7","username7","name7"))
+        userToBeAdded.add(UserEntity("uid8","username8","name8"))
+        userToBeAdded.add(UserEntity("uid9","username9","name9"))
```
Comment (Mathematou) [inline app/src/test/java/com/github/sdpteam15/polyevents/database/objects/UserDatabaseTest.kt:334]: Do we need that many test users? ^^

**Review by Mathematou — APPROVED** [review]

File: app/src/main/java/com/github/sdpteam15/polyevents/database/objects/UserDatabase.kt [hunk 2]
```diff
@@ -54,6 +57,19 @@ class UserDatabase(private val db: DatabaseInterface) : UserDatabaseInterface {
...
+                        .get()
+        ){
+            users.clear()
+            val us = it.documents.map {
+                UserAdapter.fromDocument(it.data!!, it.id)
+            }
+            users.addAll(us, this)
+        }
+    }
+
```
Comment (Rygone) [inline app/src/main/java/com/github/sdpteam15/polyevents/database/objects/UserDatabase.kt:72]:
```suggestion
    override fun updateUserInformation(
        user: UserEntity,
        userAccess: UserProfile?
    ) = db.setEntity(
        user,
        user.uid,
        USER_COLLECTION
    )
```

File: app/src/main/java/com/github/sdpteam15/polyevents/database/objects/UserDatabase.kt [hunk 3]
```diff
@@ -54,6 +57,19 @@ class UserDatabase(private val db: DatabaseInterface) : UserDatabaseInterface {
...
+                firestore!!.collection(USER_COLLECTION.value)
+                        .get()
+        ){
+            users.clear()
+            val us = it.documents.map {
+                UserAdapter.fromDocument(it.data!!, it.id)
+            }
+            users.addAll(us, this)
+        }
+    }
```
Comment (Rygone) [inline app/src/main/java/com/github/sdpteam15/polyevents/database/objects/UserDatabase.kt:71]:
```suggestion
    override fun getListAllUsers(users: ObservableList<UserEntity>, userAccess: UserProfile?) =
        db.getListEntity(
            users,
            null,
            null,
            USER_COLLECTION
        )
```

**Review by Rygone — APPROVED** [review]

File: app/src/main/res/layout/activity_user_management_list.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".admin.UserManagementListActivity">
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recycler_view_user"
```
[the remark this answers is not in the document]
Comment (AlessioVerardo) [reply app/src/main/res/layout/activity_user_management_list.xml:10]: Yep, you're right

[the remark this answers is not in the document]
Comment (AlessioVerardo) [reply app/src/main/res/layout/activity_user_management_list.xml:10]: Yep, you're right

File: app/src/test/java/com/github/sdpteam15/polyevents/model/EventTest.kt [hunk 5]
```diff
@@ -45,6 +45,27 @@ class EventTest {
...
+        )
+        assertEquals(event.eventId, null)
+        assertEquals(event.eventName, null)
+        assertEquals(event.description, null)
+        assertEquals(event.organizer, null)
+        assertEquals(event.zoneName, null)
+        assertEquals(event.icon, null)
+        assertEquals(event.startTime, null)
+        assertEquals(event.endTime, null)
+    }
```
[the remark this answers is not in the document]
Comment (AlessioVerardo) [reply app/src/test/java/com/github/sdpteam15/polyevents/model/EventTest.kt:68]: I tried, but it doesn't and I forgot to remove them :) Nice point!

File: app/src/test/java/com/github/sdpteam15/polyevents/database/objects/UserDatabaseTest.kt (hunk 1 above)
Comment (AlessioVerardo) [reply app/src/test/java/com/github/sdpteam15/polyevents/database/objects/UserDatabaseTest.kt:334]: Why not, I mean I push Ctrl+D too many times xD

Comment (AlessioVerardo) [reply app/src/test/java/com/github/sdpteam15/polyevents/database/objects/UserDatabaseTest.kt:334]: Btw, I removed this test

**Pull request merged**